ui/lib/apollo.ts

```diff
@@ -1,18 +1,33 @@
 import { ApolloClient, HttpLink, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
+import { onError } from '@apollo/client/link/error'
 import { concatPagination } from '@apollo/client/utilities'
 import merge from 'deepmerge'
 import isEqual from 'lodash/isEqual'
 import { useMemo } from 'react'
 
 let apolloClient: ApolloClient<NormalizedCacheObject>
 
+const httpLink = new HttpLink({
+  uri: '/api/graphql',
+  credentials: 'same-origin', // Additional fetch() options like `credentials` or `headers`
+})
+
+const logoutLink = onError(({ graphQLErrors }) => {
+  if (graphQLErrors) {
+    for (const err of graphQLErrors) {
+      if (err.message.includes('permission denied for')) {
+        if (typeof window !== 'undefined') {
+          window.location.href = '/login?lostSession=true'
```

Comment:
should we use `next/router` here instead?

Reply:
Done [ab457e9](https://github.com/mergestat/fuse/pull/334/commits/ab457e9d1aad3416096942ee53b5eff1d41fe9d6)